Translate how-to-make-a-reproducible-test-case (#30)
gatsbyjs/gatsby-pt-BR: docs/contributing/how-to-make-a-reproducible-test-case.md

---

> title: Como fazer How to Make a Minimal Reproduction

Reviewer: Acho que tu esqueceu uma partezinha aqui

---

> ## O que é um caso de teste produtível?

Reviewer: A idéia é "caso de teste REPRODUZÍVEL"

> Um caso de teste repoduzível é um pequeno website construído com Gatsby que demonstre o problema - frequementemente este problema é causado por um bug no Gatsby ou em um plugin do Gatsby. O seu teste de caso produtível deve conter o conjunto mínimo de funcionalidades necessárias para demonstrar o bug claramente.

Reviewer: Na primeira linha, usou reproduzível corretamente. Porém, na segunda frase, acho que ficou estranho. Sugiro:
```suggestion
Um caso de teste repoduzível é um pequeno website construído com Gatsby que demonstre o problema - frequementemente este problema é causado por um bug no Gatsby ou em um plugin do Gatsby. O seu caso de teste reproduzível deve conter um conjunto mínimo de funcionalidades necessárias, tal que seja possível demonstrar o bug claramente.
```

Reviewer 2: prePRO... nao rePO

> Um caso de teste repoduzível é um pequeno website construído com Gatsby que demonstre o problema - frequementemente este problema é causado por um bug no Gatsby ou em um plugin do Gatsby. O seu teste de caso reproduzível deve conter o conjunto mínimo de funcionalidades necessárias para demonstrar o bug claramente.

Reviewer 3: ```suggestion
Um caso de teste repoduzível é um pequeno website construído com Gatsby que demonstre um problema - frequementemente esse problema é causado por um bug no Gatsby ou em um plugin do Gatsby. O seu teste de caso reproduzível deve conter o conjunto mínimo de funcionalidades necessárias para demonstrar o bug claramente.
```

> Um caso de teste repoduzível é um pequeno website construído com Gatsby que demonstre um problema - frequementemente esse problema é causado por um bug no Gatsby ou em um plugin do Gatsby. O seu teste de caso reproduzível deve conter o conjunto mínimo de funcionalidades necessárias para demonstrar o bug claramente.

Reviewer 3: ```suggestion
Um caso de teste reproduzível é um pequeno website construído com Gatsby que demonstre um problema - frequentemente esse problema é causado por um bug no Gatsby ou em um plugin do Gatsby. O seu teste de caso reproduzível deve conter o conjunto mínimo de funcionalidades necessárias para demonstrar o bug claramente.
```

> ## Por que você deveria criar um caso de teste produtível?

Reviewer: Mesmo coisa, sugiro REPRODUZÍVEL

> Um caso de teste reprodutível permite que você isole a causa de um problema, o que é o primeiro passo na direção de corrigi-lo!

Reviewer: ```suggestion
Um caso de teste reproduzível permite que você isole a causa de um problema, o que é o primeiro passo na direção de corrigi-lo!
```

> Um caso de teste rereproduzível permite que você isole a causa de um problema, o que é o primeiro passo na direção de corrigi-lo!

Reviewer 3: ```suggestion
Um caso de teste reproduzível permite que você isole a causa de um problema, o que é o primeiro passo na direção de corrigi-lo!
```

A [parte mais importante de se reportar um bug](https://developer.mozilla.org/en-US/docs/Mozilla/QA/Bug_writing_guidelines#Writing_precise_steps_to_reproduce) é descrever extamante os passos necessários para reproduzi-lo.

> 

Reviewer 2: exatamente

> Um caso de teste reprodutível é uma excelente maneira de compartilhar as condições específicas que causam o bug. O seus caso de test reprodutível é a melhor maneira de ajudar as pessoas que querem ajudar _você_.

Reviewer: ```suggestion
Um caso de teste reproduzível é uma excelente maneira de compartilhar as condições específicas que causam o bug. O seu caso de teste reproduzível é a melhor maneira de ajudar as pessoas que querem ajudar _você_.
```

> Um caso de teste reproduzível é uma excelente maneira de compartilhar as condições específicas que causam o bug. Seu caso de test reproduzível é a melhor maneira de ajudar as pessoas que querem ajudar _você_.

Reviewer 3: ```suggestion
Um caso de teste reproduzível é uma excelente maneira de compartilhar as condições específicas que causam o bug. Seu caso de teste reproduzível é a melhor maneira de ajudar as pessoas que querem ajudar _você_.
```

> Um caso de teste reproduzível é uma excelente maneira de compartilhar as condições específicas que causam o bug. O seus caso de test rereproduzível é a melhor maneira de ajudar as pessoas que querem ajudar _você_.

Reviewer 3: ```suggestion
Um caso de teste reproduzível é uma excelente maneira de compartilhar as condições específicas que causam o bug. Seu caso de teste reproduzível é a melhor maneira de ajudar as pessoas que querem ajudar _você_.
```

> ## Passos para criar um caso de teste rereproduzível

Reviewer 3: ```suggestion
## Passos para criar um caso de teste reproduzível
```

> - Crie um novo website com Gatbsy a partir de um starter, o `hello-world` starter oficial é um execelente exemplo de caso base ou ponto de partida: `gatsby new bug-repro https://github.com/gatsbyjs/gatsby-starter-hello-world`

Reviewer 2: excelente

> - Crie um novo website com Gatbsy a partir de um _starter_, o `hello-world` _starter_ oficial é um excelente exemplo de caso base ou ponto de partida: `gatsby new bug-repro https://github.com/gatsbyjs/gatsby-starter-hello-world`

Reviewer 3: ```suggestion
- Crie um novo website com Gatbsy a partir de um _starter_, o _starter_ oficial `hello-world` é um excelente exemplo de caso base ou ponto de partida: `gatsby new bug-repro https://github.com/gatsbyjs/gatsby-starter-hello-world`
```

> - Crie um novo website com Gatbsy a partir de um _starter_, o `hello-world` _starter_ oficial é um execelente exemplo de caso base ou ponto de partida: `gatsby new bug-repro https://github.com/gatsbyjs/gatsby-starter-hello-world`

Reviewer 3: ```suggestion
- Crie um novo website com Gatbsy a partir de um _starter_, o `hello-world` _starter_ oficial é um excelente exemplo de caso base ou ponto de partida: `gatsby new bug-repro https://github.com/gatsbyjs/gatsby-starter-hello-world`
```

> - Adicione os plugins do Gatsby que estejam relacionados com o problema. Por exemplo, se você estiver tendo problemas com o Gatsby MDX você deve instalar e configurar [`gatsby-plugin-mdx`](https://www.gatsbyjs.org/packages/gatsby-plugin-mdx/). Apenas adicione os plugins que são necessários para a demonstração do problema.

Reviewer 3: ```suggestion
- Adicione os plugins do Gatsby que estejam relacionados com o problema. Por exemplo, se você estiver tendo problemas com o Gatsby MDX você deve instalar e configurar o [`gatsby-plugin-mdx`](https://www.gatsbyjs.org/packages/gatsby-plugin-mdx/). Apenas adicione os plugins que são necessários para a demonstração do problema.
```

> - Adicione os plugins do Gatsby que estejam relacionados com o problema. Por exemplo, se você estiver tendo problemas com o Gatsby MDX você deve instalar e configurar [`gatsby-plugin-mdx`](https://www.gatsbyjs.org/packages/gatsby-plugin-mdx/). Apenas adicione os plugins que são necessários para demonstrar o problema.

Reviewer 3: ```suggestion
- Adicione os plugins do Gatsby que estejam relacionados com o problema. Por exemplo, se você estiver tendo problemas com o Gatsby MDX você deve instalar e configurar o [`gatsby-plugin-mdx`](https://www.gatsbyjs.org/packages/gatsby-plugin-mdx/). Apenas adicione os plugins que são necessários para a demonstração do problema.
```

> - Adicione o código necessário para recriar o erro que você notou.

Reviewer 2: notou -> observou

> - Publique o código (a sua conta do GitHub é um bom lugar para fazer isso) e então adicione o link para ele quando estiver [criando uma issua](/contributing/how-to-file-an-issue/).

Reviewer: Com underline: `criando uma _issue_`

> ## Locais para desenvolver um caso de teste reprodutível

Reviewer 2: reproduZível (para manter o padrao)

> ## Locais para desenvolver um caso de teste rereproduzível

Reviewer 3: ```suggestion
## Locais para desenvolver um caso de teste reproduzível
```

> - Localmente com um start: Você pode começar com um [Starter](/docs/starters) localmente e então faça um build na sua própria máquina. Os starters oficiais do Gatsby [`hello-world`](https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world) ou [`default`](https://github.com/gatsbyjs/gatsby-starter-default) são ambos boas bases para um caso de teste reprodutível.

Reviewer: `_starter_`

> - Localmente com um _starter_: Você pode começar com um [_starter_](/docs/starters) localmente para então fazer um build na sua própria máquina. Os _starters_ oficiais do Gatsby [`hello-world`](https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world) ou [`default`](https://github.com/gatsbyjs/gatsby-starter-default) são ambos boas bases para um caso de teste reproduzível.

Reviewer 3: ```suggestion
- Localmente com um _starter_: Você pode começar com um [_starter_](/docs/starters) localmente para então fazer um build na sua própria máquina. Os _starters_ oficiais do Gatsby [`hello-world`](https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world) ou [`default`](https://github.com/gatsbyjs/gatsby-starter-default) ambos são boas bases para um caso de teste reproduzível.
```

> - Localmente com um _starter_: Você pode começar com um [_starter_](/docs/starters) localmente e então faça um build na sua própria máquina. Os _starters_ oficiais do Gatsby [`hello-world`](https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world) ou [`default`](https://github.com/gatsbyjs/gatsby-starter-default) são ambos boas bases para um caso de teste rereproduzível.

Reviewer 3: ```suggestion
- Localmente com um _starter_: Você pode começar com um [_starter_](/docs/starters) localmente para então fazer um build na sua própria máquina. Os _starters_ oficiais do Gatsby [`hello-world`](https://github.com/gatsbyjs/gatsby/tree/master/starters/hello-world) ou [`default`](https://github.com/gatsbyjs/gatsby-starter-default) são boas bases para um caso de teste reproduzível.
```

- Hospede no CodeSandbox: Você pode desenvolver um website com Gatsby diretamente do seu navegador com o CodeSandbox usando o [template Gatsby](https://codesandbox.io/s/github/gatsbyjs/gatsby-starter-default) deles. CodeSandbox também hospeda o seu website automaticamente, o que pode ser útil para demonstrar o comportamento do seu site.

> ## Benefícios de um caso de teste reprodutível

Reviewer 2: reproduZível (para manter o padrao)

> - Menor superfície de contato: Removendo tudo menos o erro você não tem que cavar para encontrar o bug.

Reviewer: A vírgula do arquivo original em inglês cairia bem em português também.

> - Menor superfície de contato: Removendo tudo, menos o erro, você não tem que cavar para encontrar o bug.

Reviewer 3: ```suggestion
- Menor superfície de contato: Ao remover tudo, menos o erro, faz com que você não tenha que cavar para encontrar o bug.
```

O que tu acha dessa sugestão?

Author: Acho que melhora a leitura, acho que daria pra aplicar isso em outros lugares da doc, mas é difícil identificar sempre.

> - Sem necessidade de publicar código privado: Talvez você não consiga publicar o seu website principal (por diversas razões). Recriar uma pequena parte dele como um caso de teste reprodutível permite que você demonstre publicamente o problema sem expor qualquer código secreto/privado.

Reviewer 2: No need to ->  não há necessidade

> - Sem necessidade de publicar código privado: Talvez você não consiga publicar o seu website principal (por diversas razões). Recriar uma pequena parte dele como um caso de teste reprodutível permite que você demonstre publicamente o problema sem expor qualquer código secreto/privado.

Reviewer 2: secrete code -> código sigiloso

Author: Eu acho que a intenção é dizer que o código é de domínio privado, ou seja, a empresa que a pessoa trabalha detém o direito do código. Por essa razão achei que `privado` cairia melhor, o que acha de usar só `privado`?

> - Não há necessidade de publicar código privado: Talvez você não consiga publicar o seu website principal (por diversas razões). Recriar uma pequena parte dele como um caso de teste rereproduzível permite que você demonstre publicamente o problema sem expor qualquer código privado.

Reviewer 3: ```suggestion
- Não há necessidade de publicar código privado: Talvez você não consiga publicar o seu website principal (por diversas razões). Recriar uma pequena parte dele como um caso de teste reproduzível permite que você demonstre publicamente o problema sem expor qualquer código privado.
```

> - Prova da existência do bug: Algumas vezes um bug é causado por alguma combinação de configurações na sua máquina. Um caso de teste produtível permite que os contribuidores baixem o seu build e testem ele em suas máquinas também. Isso ajuda a verificar e restringir a casua do problema.

Reviewer: casua -> causa

> - Prova da existência do bug: Algumas vezes um bug é causado por alguma combinação de configurações na sua máquina. Um caso de teste produtível permite que os contribuidores baixem o seu build e testem ele em suas máquinas também. Isso ajuda a verificar e restringir a casua do problema.

Reviewer 2: reproduzível (para manter o padrao)

> - Receba ajuda para corrigir o seu bug: Se outra pessoa consegue reproduzir o problema, frequentemente  eles tem uma boa chance de corrigi-lo. É quase impossível corrigir um bug sem primeiro conseguir reproduzilo.

Reviewer: reproduzilo -> reproduzi-lo

Reviewer 2: eles -> ela

Reviewer 3: ```suggestion
- Receba ajuda para corrigir o seu bug: Se outra pessoa consegue reproduzir o problema, Normalmente ela tem uma boa chance de corrigi-lo. É quase impossível corrigir um bug sem primeiro conseguir reproduzi-lo.
```
